ember-resources/src/util/function.ts

```diff
@@ -92,28 +92,78 @@ export function trackedFunction<Return>(...passedArgs: UseFunctionArgs<Return>)
     fn = passedArgs[2];
   }
 
-  return resource<State<Return>>(context, (hooks) => {
-    let state = new State(initialValue);
+  return buildResource(context, fn, initialValue);
+}
 
-    (async () => {
-      try {
-        let notQuiteValue = fn(hooks);
-        let promise = Promise.resolve(notQuiteValue);
+function createPropertyValue(getValue) {
+  const propertyValue = function () {
+    propertyValue.state = getValue();
+    return propertyValue.state;
+  };
+
+  Object.defineProperty(propertyValue, "value", {
```

Comment:
This likely won't be TS safe -- any reason to not use a class?

Reply:
I'm not sure how you can define a class which can then be called as a function. Just out of interest, do you know how to do this?